### PR TITLE
paste: fix pasting images from excalidraw

### PR DESCRIPTION
**Pull request by mimecuvalo**
I didn't dig into when/where this regressed but we've changed a lot in this logic so I'm not surprised.

### Change type

- [x] `bugfix`
- [ ] `improvement`
- [ ] `feature`
- [ ] `api`
- [ ] `other`

### Release notes

- Pasting: fix image pasting from Excalidraw.

### CODE REVIEW

**Comment by ds300** [comment]
this breaks pretty often. i wonder whether we should make it part of our qa script. files too.

**Review by SomeHats — COMMENTED** [review]
we have a very small number of automated test cases for excalidraw pasting - you can add excalidraw clipboard json to `excalidraw-test-fixtures` and we'll automatically generate a snapshot test for the resulting tldraw document. Could we add one that involves pasting an image?

**Comment by mimecuvalo** [comment]
> we have a very small number of automated test cases for excalidraw pasting - you can add excalidraw clipboard json to excalidraw-test-fixtures and we'll automatically generate a snapshot test for the resulting tldraw document. Could we add one that involves pasting an image?

Actually, I should be clearer, this isn't _exactly_ specific to Excalidraw. It's just anything that has a `data:image` on the clipboard. But yeah, I'll add something to the fixtures.

**Comment by steveruizok** [comment]
Nice one! I'm loving these fix PRs.

File: packages/tldraw/src/lib/ui/hooks/clipboard/__snapshots__/pasteExcalidrawContent.test.tsx.snap [hunk 1]
```diff
@@ -148,6 +148,63 @@ exports[`pasteExcalidrawContent test fixtures bound-arrows.json 1`] = `
...
+  "asset:9": {
+    "id": "asset:9",
+    "meta": {},
+    "props": {
+      "fileSize": undefined,
+      "h": 1,
+      "isAnimated": false,
+      "mimeType": "image/svg+xml",
+      "name": "cbB-uDKihdtuW5FkgGUGi",
+      "src": null,
```
[the remark this answers is not in the document]
Comment (mimecuvalo) [reply packages/tldraw/src/lib/ui/hooks/clipboard/__snapshots__/pasteExcalidrawContent.test.tsx.snap:162]: to be fair, this test doesn't really test this issue heh. i added this to the snapshot but the fix/bug is actually in `defaultExternalContentHandlers.ts`

**Comment by MitjaBezensek** [comment]
Not sure why but getting some CSP issues when trying this locally (working with the preview link though):
image

**Comment by mimecuvalo** [comment]
> Not sure why but getting some CSP issues when trying this locally (working with the preview link though): image

ah, this just needs to be merged with the latest on main. i'll do that now

**Review by MitjaBezensek — APPROVED** [review]

**Pull request merged**